gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -616,8 +638,23 @@ protected static DiffPathSet fullPathDiff(HiveLocationDescriptor sourceLocation,
       if (desiredTargetExistingPaths.containsKey(newPath)) {
         // If the file exists at the destination, check whether it should be replaced, if not, no need to copy
         FileStatus existingTargetStatus = desiredTargetExistingPaths.get(newPath);
-        if (!helper.shouldReplaceFile(existingTargetStatus, sourcePath)) {
-          shouldCopy = false;
+
+        Comparable srcVer = sourceLocation.versionStrategy.get().getVersion(sourcePath.getPath());
+        Comparable dstVer = desiredTargetLocation.versionStrategy.get().getVersion(existingTargetStatus.getPath());
+
+        // destination has higher version, skip the copy
+        if (srcVer.compareTo(dstVer) <= 0) {
+          if (!helper.isEnforceFileSizeMatch() || existingTargetStatus.getLen() == sourcePath.getLen()) {
```

Comment:
It seems that if `isEnforceFileSizeMatch` is true, then the data will never be copied?

Reply:
Read the logic incorrectly, ignore this comment.